sessiontxn/isolation/repeatable_read.go

```diff
@@ -179,24 +182,58 @@ func (p *PessimisticRRTxnContextProvider) AdviseOptimizeWithPlan(val interface{}
 		plan = execute.Plan
 	}
 
-	mayOptimizeForPointGet := false
-	if v, ok := plan.(*plannercore.PhysicalLock); ok {
-		if _, ok := v.Children()[0].(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
+	p.optimizeForNotFetchingLatestTS = optimizeForNotFetchingLatestTS(plan, false)
+
+	return nil
+}
+
+// optimizeForNotFetchingLatestTS searches for optimization condition recursively
+// Note: For point get and batch point get (name it plan), if one of the ancestor node is update/delete/physicalLock,
+// we should check whether the plan.Lock is true or false. See comments in needNotToBeOptimized.
+// flag = true means one of the ancestor node is update/delete/physicalLock.
+func optimizeForNotFetchingLatestTS(plan plannercore.Plan, flag bool) bool {
+	switch v := plan.(type) {
+	case *plannercore.PointGetPlan, *plannercore.BatchPointGetPlan:
+		if needNotToBeOptimized(plan, flag) {
+			return false
 		}
-	} else if v, ok := plan.(*plannercore.Update); ok {
-		if _, ok := v.SelectPlan.(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
+		return true
+	case plannercore.PhysicalPlan:
+		if v.Children() == nil {
+			return false
 		}
-	} else if v, ok := plan.(*plannercore.Delete); ok {
-		if _, ok := v.SelectPlan.(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
+		allChildrenArePointGet := true
+		_, isPhysicalLock := v.(*plannercore.PhysicalLock)
+		for _, p := range v.Children() {
+			allChildrenArePointGet = allChildrenArePointGet && optimizeForNotFetchingLatestTS(p, isPhysicalLock || flag)
 		}
+		return allChildrenArePointGet
+	case *plannercore.Update:
+		return optimizeForNotFetchingLatestTS(v.SelectPlan, true)
+	case *plannercore.Delete:
+		return optimizeForNotFetchingLatestTS(v.SelectPlan, true)
+	case *plannercore.Insert:
+		return v.SelectPlan == nil
```

Comment:
If `v.SelectPlan` is point get, we can also Optimize it.

Reply:
It is not optimized originally, so, it's better not to optimize it in this PR.